src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -169,21 +261,24 @@ trait EntityComponent {
       unmarshalEntitiesWithIds(entityAttributeAction).map(_.map { case (id, entity) => entity })
     }
 
-    def unmarshalEntitiesWithIds(entityAttributeAction: ReadAction[Seq[EntityAndAttributesRawSqlQuery.EntityAndAttributesResult]]): ReadAction[Map[Long, Entity]] = {
-      entityAttributeAction.map { entityAttributeRecords =>
-        val allEntityRecords = entityAttributeRecords.map(_.entityRecord).toSet
+    def unmarshalEntitiesWithIds(entityAttributeAction: ReadAction[Seq[EntityAndAttributesRawSqlQuery.EntityAndAttributesResult]]): ReadAction[mutable.LinkedHashMap[Long, Entity]] = {
+      entityAttributeAction.map(unmarshalEntitiesWithIds)
+    }
 
```

Comment:
these changes to use LinkedHashMap are to preserve ordering from the sql results